parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -2479,6 +2495,22 @@ opt_block_args_tail:
                       $$ = block;
                       driver.lex.context.pop();
                     }
+                  // Error-recovery case for typos like "foo {&:bar}"
+                | tLCURLY
+                    {
+                      driver.lex.context.push(Context::State::BLOCK);
+                    }
```

Comment:
this causes a reduce/reduce conflict, doesn't it? with the midrule action for the case immediately above?

Reply:
I thought it would, but oddly it doesn't seem to.

Reply:
interesting. to be explicit and to ward against this being a problem in the future, what do you think about factoring that out into a proper production, like

```
lcurly_block_start : tLCURLY { push(...) }
```

and then using the lcurly_block_start non-terminal in both places?